packages/ingest-mongodb-public/src/sources/WebDataSource.ts

```diff
@@ -0,0 +1,352 @@
+  "https://www.mongodb.com/resources/basics/unified-modeling-language",
+  "https://www.mongodb.com/resources/basics/ai-hallucinations",
+  "https://www.mongodb.com/resources/basics/ai-in-finance",
+  "https://www.mongodb.com/resources/basics/what-is-stream-processing",
+  "https://www.mongodb.com/resources/basics/cognitive-search",
+  "https://www.mongodb.com/resources/basics/foundation-models",
+  "https://www.mongodb.com/resources/basics/ai-stack",
+  "https://www.mongodb.com/resources/products/compatibilities/kubernetes",
+  "https://www.mongodb.com/resources/products/capabilities/stored-procedures",
+  "https://www.mongodb.com/resources/basics/cloud-explained/iaas-infrastructure-as-a-service",
+  "https://www.mongodb.com/resources/basics/databases/cloud-databases/cloud-migration",
+  "https://www.mongodb.com/resources/basics/unstructured-data/tools",
+  "https://www.mongodb.com/resources/basics/system-of-insight",
+  "https://www.mongodb.com/resources/basics/databases/middleware",
+  "https://www.mongodb.com/resources/basics/disaster-recovery",
+  "https://www.mongodb.com/resources/basics/cloud-explained/business-intelligence-bi-tools",
+  "https://www.mongodb.com/resources/basics/observability",
+  "https://www.mongodb.com/resources/basics/role-based-access-control",
+  "https://www.mongodb.com/resources/basics/real-time-payments",
+  "https://www.mongodb.com/resources/basics/databases/cloud-databases/free-cloud-database",
+];
+
+export function makeWebDataSource({
+  maxPages,
+}: {
+  maxPages?: number; // for testing
+}): DataSource {
+  return {
+    name: "mongodb-web",
+    async fetchPages() {
+      let browser;
+      try {
+        const { page: puppeteerPage, browser: b } = await makePuppeteer();
+        browser = b;
+        // Use the sitemap to get the urls belonging to the directories we want every page from
+        const completeDirectoryUrls = await getCompleteDirectoriesFromSitemap(
+          sitemapURL
+        );
+        // Get urls from the list provided by web team
+        const urls = [...completeDirectoryUrls, ...individualUrls];
+        const pages: Page[] = [];
+        for await (const [index, url] of urls.entries()) {
+          if (maxPages && index >= maxPages) {
+            break;
+          }
+          pages.push(
+            await scrapePage({
+              url,
+              puppeteerPage,
+            })
+          );
+        }
+        return pages;
+      } finally {
+        await browser?.close();
+      }
+    },
+  };
+}
+
+export async function getUrlsFromSitemap(
+  sitemapURL: string
+): Promise<string[]> {
+  const response = await fetch(sitemapURL);
+  const sitemap = await response.text();
+  const parser = new xml2js.Parser();
+  const parsedXML = await parser.parseStringPromise(sitemap);
+  return parsedXML.urlset.url.map((url: { loc: string[] }) => url.loc[0]);
+}
+
+export async function getCompleteDirectoriesFromSitemap(
```

Comment:
```suggestion
async function getCompleteDirectoriesFromSitemap(
```

you dont use this func outside the test, dont export